rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -175,10 +163,15 @@ std::shared_ptr<GenericSubscription>
 Recorder::create_subscription(
   const std::string & topic_name, const std::string & topic_type, const rclcpp::QoS & qos)
 {
+  Rosbag2QoS subscription_qos(qos);
+  if (qos_profile_overrides_[topic_name]) {
```

Comment:
rather than putting this logic here, I think we should update `common_qos_or_fallback` to be the single source of truth for what QoS to use for a topic, that way the logic isn't distributed.

Maybe rename it as `subscription_qos_for_topic` and update the docstring to clarify that it does a priority ordering of "override, common qos, fallback"